packages/core/src/browser/compressible-tree/tree-compression.ts

```diff
@@ -0,0 +1,90 @@
+/********************************************************************************
+ * Copyright (c) 2021 SAP SE or an SAP affiliate company and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { ExpandableTreeNode } from '../tree/tree-expansion';
+import { TreeNode, CompositeTreeNode } from '../tree';
+
+/**
+ * Represents a tree node that is able to be compressed into
+ * its parent node depending on its `compressed` state.
+ */
+export interface CompressibleTreeNode extends ExpandableTreeNode {
+    /**
+     * Indicates whether the tree node can be compressed into its parent node
+     */
+    compressible: boolean;
+}
```

Comment:
As far as I can tell, this interface is no longer really used. There's code in `compressible-tree` that manipulates this field, but no code that checks it elsewhere. I think the interface can be removed, and the code below that returns `node is CompressibleTreeNode` or the like can be replaced with `node is CompositeTreeNode`.

Reply:
@colin-grant-work I use it here:
 ``` `/** Determines whether the node is a compressed child */
    export function isCompressionChild(node: Object | undefined): node is CompressibleTreeNode {
        return is(node) && node.compressible && isCompressibleParent(node.parent);
    }` ```
([A link to the code](https://github.com/EstherPerelman/theia/blob/f44787127aacbfea20510f9097093f4d4b935253/packages/core/src/browser/compressible-tree/tree-compression.ts#L46-L49))
This what does all the work of the compressible node...

Reply:
As I've [said before](https://github.com/eclipse-theia/theia/pull/10268#discussion_r738710469), everything you need to know about whether a node is compressible can be determined from its features and those of its parent; you don't need the `compressible` field. A node is a compressed child (i.e. a non-initial node in a compressed row) under the following conditions:
 - its parent is a composite tree node
 - its parent has a single child (the node itself)
 - the node itself is a composite tree node